app/models/note_comment.rb

```diff
@@ -25,7 +25,7 @@
 
 class NoteComment < ActiveRecord::Base
   belongs_to :note, :foreign_key => :note_id, :touch => true
-  belongs_to :author, :class_name => "User", :foreign_key => :author_id
+  belongs_to :author, :class_name => "User", :foreign_key => :author_id, :counter_cache => :notes_count
```

Comment:
What was the reason for overriding the name here? Would the default of `note_comments_count` not be more accurate?

Reply:
I think `note_comments_count` is less accurate because we're not counting the number of comments but rather the number of authored notes. So `note_comments_count` isn't exactly right because a user can comment multiple times on a note but maybe `note_authored_count` is a better name?

Reply:
Are you sure it's counting the number of notes? I'm pretty sure that's going to count the total number of comments isn't it?

It's on the relationship between user and note comments, and the documentation explains:

> The counter cache is incremented when an object of this class is created and decremented when it's destroyed. 

So it will count every object of class `NoteComment` belonging to a given user.

Reply:
Ah that's right, I misunderstood.

Since there's no distinction of an author in the `Note` class, do you have a recommendation for how to count just the first comment on a note? That logic would have to be within the `NoteComment` class right?

Reply:
I don't think you can using the builtin counter caches, or at least I tried various things without any success.

The general suggestion seems to be https://github.com/magnusvk/counter_culture which allows for conditional counters.